packages/labs/src/scoped-registry/scoped-registry-mixin.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * @license
+ * Copyright (c) 2018 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {LitElement} from 'lit-element';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type LitElementConstructor = new (...args: any[]) => LitElement;
+
+export type ScopedElementsMap = {
+  [key: string]: typeof HTMLElement;
+};
+
+export function UseScopedRegistry<SuperClass extends LitElementConstructor>(
+  superclass: SuperClass
+): SuperClass {
+  return class ScopedRegistryMixin extends superclass {
+    registry?: CustomElementRegistry;
+
+    /**
+     * Obtains the scoped elements definitions map
+     */
+    static get scopedElements(): ScopedElementsMap {
+      return {};
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    constructor(...args: any[]) {
+      super(...args);
+
+      this.registry = new CustomElementRegistry();
+
+      const {scopedElements} = this.constructor as typeof ScopedRegistryMixin;
+
+      Object.entries(scopedElements).forEach(([tagName, klass]) =>
+        this.registry?.define(tagName, klass)
+      );
+    }
+
+    createRenderRoot() {
+      this.renderOptions.shadowRoot = this.attachShadow({
+        mode: 'open',
+        // @ts-expect-error: customElements not yet in ShadowRootInit type
+        customElements: this.registry,
+      });
+
+      return this.renderOptions.shadowRoot;
```

Comment:
you can just put the `return` keyword in front of the assignment above

Reply:
Right. Thanks!

Reply:
Done